### ISSUE TITLE
Doesnt work! Please Help onRM Bridge LAN Readme.md 

### ISSUE THREAD

**Issue by jimmysoegiarto** [beckyricha/Broadlink-RM-SmartThings-Alexa#2]
Dear beckyricha

I really appreciate what you have done.  I think it would be awsome to have it running.

But unfortunately, I have followed every instructions and steps on your "RM Bridge LAN Readme.md" And I have tried it 3 times.  It doesnt work!!

I can see my device show up on my Smartthings App on my Android, but when I press the button on/off (green icon) it didnt do anything.

Please find attached
![photo on 8-12-16 at 8 20 pm](https://cloud.githubusercontent.com/assets/20991055/17623537/c3cbf47c-60ca-11e6-915b-5cd4dd503e76.jpg)

Eventhough it works on the "test the result" on RM Bridge site

![screen shot 2016-08-12 at 8 26 14 pm](https://cloud.githubusercontent.com/assets/20991055/17623630/2acf00c4-60cb-11e6-9997-ed165e4983c6.png)

Please let me know what am I missing. I really appreciate it

Thank you

**Comment by jimmysoegiarto** [comment]
![photo on 8-12-16 at 8 20 pm](https://cloud.githubusercontent.com/assets/20991055/17623474/69e9c7ea-60ca-11e6-8216-74d698dd43f6.jpg)

AS you can see, the "coba" devices is in my Smartthings App, but it doesnt do anything when you press on/off

**Comment by beckyricha** [comment]
Hi. I will try to help if i can but will need more info than this to help you.  It's impossible to tell what may not be working based only on following the instructions and not having success.  

Please let me know exactly which version of the app you are using (whether via the local hub control or network service, and which Android bridging app), what interim troubleshooting steps your have already tried, etc. For instance what sort of device are you trying to control?   does the device work using the broadlink software directly? Using the Android bridge directly? If you look at the device properties in the smart things ide sure it display the correct network information, remote codes and custom device type?  

I'll try to help if i can but please test the various pieces separately to try to help narrow down where the issue might be.  Smart things occasionally also has some issues that affect it but that's less common so we can look for some other things first.

Becky



On August 12, 2016 9:18:11 AM EDT, jimmysoegiarto <notifications@github.com> wrote:
>Dear beckyricha
>I have followed every instructions and steps on your "RM Bridge LAN
>Readme.md"
>I have tried it 3 times.
>
>I can see my device show up on my Smartthings App on my Android, but
>when I press the button on/off (green icon) it didnt do anything.
>
>Please let me know what am I missing.
>
>Thank you
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/beckyricha/Broadlink-RM-SmartThings-Alexa/issues/2

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by beckyricha** [comment]
Also please let me know what version hub you are using, and whether any errors appear when you run the code from the ide. In the phone app  you won't see the error messages so you need to try it from the ide to see it. This will be on the same page where you can view the device handler code (not the smart app). Just choose on the right side your device and select install. Then you can turn it off and on from the same web page and see what it did. Any errors will appear below the code in the log.



On August 12, 2016 9:18:11 AM EDT, jimmysoegiarto <notifications@github.com> wrote:
>Dear beckyricha
>I have followed every instructions and steps on your "RM Bridge LAN
>Readme.md"
>I have tried it 3 times.
>
>I can see my device show up on my Smartthings App on my Android, but
>when I press the button on/off (green icon) it didnt do anything.
>
>Please let me know what am I missing.
>
>Thank you
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/beckyricha/Broadlink-RM-SmartThings-Alexa/issues/2

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by beckyricha** [comment]
Hi. I saw on github that you did add some details there that will help.  For some reason github does not send emails when the issue posted so i only had your separate note before.  

Seeing that, I can now ask better questions:
1) what happens when you enter the shortcut url into a browser?

2) please do let me know if you are using the lan or cloud version if the device handler.

3) if using the lan version, is your device connected to your hub? (I don't know of any way to see this from the app; have to go into the ide and look at the device settings).

4) if using the networked version, is port 7474 forwarded to the Android device? What happens if you enter the shortcut url into a browser using your remote ip address instead of the local one?

Sorry i didn't get a sense of what sort of user you are so if you need help with figuring out any of these questions please let me know.

Becky


On August 12, 2016 9:22:42 AM EDT, jimmysoegiarto <notifications@github.com> wrote:
>![photo on 8-12-16 at 8 20
>pm](https://cloud.githubusercontent.com/assets/20991055/17623474/69e9c7ea-60ca-11e6-8216-74d698dd43f6.jpg)
>
>AS you can see, the "coba" devices is in my Smartthings App, but it
>doesnt do anything when you press on/off
>
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/beckyricha/Broadlink-RM-SmartThings-Alexa/issues/2#issuecomment-239444656

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jimmysoegiarto** [comment]
Dear Becky, thanks for your response, here is my answers in bold

what sort of device are you trying to control?   **lgdvd**

does the device work using the broadlink software directly? **it works**

Using the Android bridge directly? **Using RM bridge apps, it works**

If you look at the device properties in the smart things ide sure it display the correct network information, remote codes and custom device type?
**Im not sure about it, how do I do this? I have include screenshots, please find link below**

1) what happens when you enter the shortcut url into a browser?

**it turns on the device**

2) please do let me know if you are using the lan or cloud version if the device handler.
3) if using the lan version, is your device connected to your hub? (I don't know of any way to see this from the app; have to go into the ide and look at the device settings).

**I use the LAN version and copy paste your code, Broadlink RM Bridge Switch LAN**

Fyi, here is my step by step including all screen shots of Smartthings IDE, RM Bridge

https://www.dropbox.com/s/oghrxdvbpd6y82t/my%20step%20by%20step%20broadlink%20smartthings.doc?dl=0

My email is jimmy.soegiarto@gmail.com  I also available on google chat if this help.
I really appreciate your help.

Thank you

**Comment by beckyricha** [comment]
Thanks. This narrows it down a lot.  I was not able to see the screen shot in the file but may not need it. I think from what you said that the issue is somewhere in the steps where the device handler was set up and the device itself was added.   I had not realized this email exchange was also posting on github.  I will email you separately as i would like some details that you may not wish to post publicly. If we find a solution useful to others then i will repost something to github for others.



On August 14, 2016 5:09:30 AM EDT, jimmysoegiarto <notifications@github.com> wrote:
>Dear Becky, thanks for your response, here is my answers in bold
>
>what sort of device are you trying to control?   **lgdvd**
>
>does the device work using the broadlink software directly? **it
>works**
>
>Using the Android bridge directly? **Using RM bridge apps, it works**
>
>If you look at the device properties in the smart things ide sure it
>display the correct network information, remote codes and custom device
>type?
>**Im not sure about it, how do I do this? I have include screenshots,
>please find link below**
>
>1) what happens when you enter the shortcut url into a browser?
>
>**it turns on the device**
>
>2) please do let me know if you are using the lan or cloud version if
>the device handler.
>3) if using the lan version, is your device connected to your hub? (I
>don't know of any way to see this from the app; have to go into the ide
>and look at the device settings).
>
>**I use the LAN version and copy paste your code, Broadlink RM Bridge
>Switch LAN**
>
>Fyi, here is my step by step including all screen shots of Smartthings
>IDE, RM Bridge
>
>https://www.dropbox.com/s/oghrxdvbpd6y82t/my%20step%20by%20step%20broadlink%20smartthings.doc?dl=0
>
>My email is jimmy.soegiarto@gmail.com  I also available on google chat
>if this help.
>I really appreciate your help.
>
>Thank you
>
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/beckyricha/Broadlink-RM-SmartThings-Alexa/issues/2#issuecomment-239663604

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jimmysoegiarto** [comment]
Hi Becky

Thank you for your respons.

I actually deleted the coba device, device handler and started from scratch again with my lgdvd.

I re-checked again as your instructed.
Now I know the problem. The smartthings hub wasnt connected to the device.

I edited and updated and now it is working fine.  

Thank you very much for your patience and your help, deeply appreciate it.

It might be helpful to update your github read me, specially when adding device to smartthings, it needs to add hub and location on the drop list.

Again, thank you so much, I owe you :)
